frontend/src/pages/clusterSettings/ClusterSettings.tsx

```diff
@@ -327,6 +346,58 @@ const ClusterSettings: React.FC = () => {
                 />
               </FormGroup>
             ) : null}
+            {dashboardConfig.spec.notebookController?.enabled ? (
+              <FormGroup fieldId="notebook-toleration" label="Notebook pod tolerations">
+                <Checkbox
+                  label="Add a toleration to notebook pods to allow them to be scheduled to tainted nodes"
+                  isChecked={notebookTolerationSettings?.enabled}
+                  onChange={() => {
+                    const newNotebookTolerationSettings = {
+                      ...notebookTolerationSettings,
+                      enabled: !notebookTolerationSettings?.enabled,
+                    };
+                    setNotebookTolerationSettings(
+                      newNotebookTolerationSettings as NotebookTolerationSettings,
+                    );
+                  }}
+                  aria-label="tolerationsEnabled"
+                  id="tolerations-enabled-checkbox"
+                  name="tolerationsEnabledCheckbox"
+                />
+                <InputGroup>
+                  <InputGroupText variant={InputGroupTextVariant.plain}>
+                    Toleration key for notebook pods:{' '}
+                  </InputGroupText>
+                  <TextInput
```

Comment:
Should we also disable the toleration input when the checkbox is unchecked? That seems like a common way when we deal with both checkbox and input working together.

Reply:
done.